### ISSUE TITLE
gossipsub: documented TODO

### ISSUE THREAD

**Issue by drHuangMHT** [libp2p/rust-libp2p#4557]
`unused_doc_comments` now generates warning.
https://github.com/libp2p/rust-libp2p/blob/38ea7ad4534565876225454a71edf91ce67adb16/protocols/gossipsub/src/types.rs#L333

**Comment by thomaseizinger** [comment]
What kind of warning? Is it easy to fix?

It is a shame that denying warnings doesn't work for rustdoc: https://github.com/libp2p/rust-libp2p/issues/3282

**Comment by drHuangMHT** [comment]
The warning is as follows:
```
warning: unused doc comment
   --> C:\Users\DrHuang\Documents\GithubProjects\rust-libp2p\protocols\gossipsub\src\types.rs:333:33
    |
333 | ...                   /// TODO, see https://github.com/libp2p/specs/pull/217
    |                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
334 | ...                   signed_peer_record: None,
    |                       ------------------------ rustdoc does not generate documentation for expression fields
    |
    = help: use `//` for a plain comment
    = note: `#[warn(unused_doc_comments)]` on by default
```
The easiest fix will be changing triple slashes to double slashes, but [#217](https://github.com/libp2p/specs/pull/217) has been resolved.

**Comment by thomaseizinger** [comment]
The relevant feature here is called "peer exchange" and is tracked here: https://github.com/libp2p/rust-libp2p/issues/2398

Could you send a PR though to fix the warning by not using a triple-slash?

**Comment by drHuangMHT** [comment]
Fixed in #4811.
